### PR TITLE
Gradle Enterprise server URL is overridable

### PR DESCRIPTION
**Pull request by alextu**
This PR adds a checkbox to the configuration UI to allow the GE URL configuration to override the values configured in projects. This may be useful for migrating projects from one server to another. In particular, it can be useful for projects that are already configured to upload scans to scans.gradle.com.

Same as already implemented in TC https://github.com/etiennestuder/teamcity-build-scan-plugin/pull/236

image

### Testing done
- integration test just testing presence of env var when setting the `enforceUrl` config flag
- acceptance test setting the `enforceUrl` flag, building a project with a preconfigured GE URL publishes to the global configured GE URL. Note that it does not test that the `allowUntrustedServer` is also overridden. This has been tested manually.

### Submitter checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -134,6 +134,8 @@ Next set the URL of the Gradle Enterprise instance to which the build scans shou
 
 image::enable-auto-injection.png[Enable auto-injection,{thumbnail}]
```
Comment (c00ler) [inline README.adoc:135]: Do we need to update the screenshot to the new one that includes the checkbox?

File: acceptance-tests/src/main/java/hudson/plugin/gradle/BuildScansInjectionSettings.java [hunk 2]
```diff
@@ -43,6 +43,18 @@ public void setGradleEnterpriseServerUrl(URI server) {
...
+    public void clickEnforceUrl() {
+        ensureConfigPage();
+
+        control(by.checkbox("Enforce Gradle Enterprise server url")).click();
+    }
+
+    public void clickAllowUntrustedServer() {
+        ensureConfigPage();
+
+        control(by.checkbox("Allow untrusted server")).click();
```
Comment (c00ler) [inline acceptance-tests/src/main/java/hudson/plugin/gradle/BuildScansInjectionSettings.java:55]: We use these 2 methods together quite often now, worth extracting the method. 

No need to do it in this PR can be done later.

File: acceptance-tests/src/test/java/hudson/plugins/gradle/GradleInjectionTest.java [hunk 3]
```diff
@@ -275,6 +275,36 @@ public void injectionWorksOnAgents() throws Exception {
         assertBuildScanPublished(build);
     }
 
+    @Test
+    public void setEnforceUrlPublishesBuildScan() throws Exception {
+        // given
+        setEnforceUrl();
+        setAllowUntrustedServer();
+        String expectedAgentVersion = "3.10.3";
```
Comment (c00ler) [inline acceptance-tests/src/test/java/hudson/plugins/gradle/GradleInjectionTest.java:283]: Can we use the latest version here?

File: src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/help-enforceUrl.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Whether to enforce the Gradle Enterprise Server URL configured in this connection over a URL configured in the project's build.
```
Comment (c00ler) [inline src/main/resources/hudson/plugins/gradle/injection/InjectionConfig/help-enforceUrl.html:2]: It must be phrased differently I think. There is no such thing as a connection. Connection is a TC-specific thing.

File: acceptance-tests/src/main/java/hudson/plugin/gradle/BuildScansInjectionSettings.java (hunk 2 above)
Comment (alextu) [reply acceptance-tests/src/main/java/hudson/plugin/gradle/BuildScansInjectionSettings.java]: done in this PR

**Review by c00ler — APPROVED** [review]

**Pull request merged**